src/main/java/org/sonarlint/intellij/promotion/PromotionProvider.kt

```diff
@@ -26,49 +26,82 @@ import com.intellij.openapi.project.Project
+
     fun subscribeToTriggeringEvents() {
         val busConnection = project.messageBus.connect()
         with(busConnection) {
             subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, object : FileEditorManagerListener {
                 override fun fileOpened(source: FileEditorManager, file: VirtualFile) {
                     if (!Settings.getSettingsFor(project).isBound && !Settings.getGlobalSettings().isPromotionDisabled) {
-                        processPromotion(source, file)
+                        processExtraLanguagePromotion(file, getSonarLintProjectNotifications(source))
+                        processAdvancedLanguagePromotion(file, getSonarLintProjectNotifications(source))
                     }
                 }
             })
         }
     }
 
-    private fun processPromotion(source: FileEditorManager, file: VirtualFile) {
-        val notifications: SonarLintProjectNotifications = SonarLintUtils.getService(
+    private fun getSonarLintProjectNotifications(source: FileEditorManager): SonarLintProjectNotifications {
+        return SonarLintUtils.getService(
             source.project,
             SonarLintProjectNotifications::class.java
         )
+    }
+
+    private fun processAdvancedLanguagePromotion(
+        file: VirtualFile,
+        notifications: SonarLintProjectNotifications,
+    ) {
+        val extension = file.extension ?: return
+
+        val language = findLanguage(extension, languagesHavingAdvancedRules)
+
+        if (language != null) {
+            showAdvancedLanguagePromotion(notifications, language.label)
+        }
+    }
 
+    private fun processExtraLanguagePromotion(file: VirtualFile, notifications: SonarLintProjectNotifications) {
         val extension = file.extension ?: return
 
-        val language = findLanguage(extension)
+        val language = findLanguage(extension, extraEnabledLanguagesInConnectedMode)
 
         if (language != null) {
-            showPromotion(notifications, language.label)
+            showWiderLanguagePromotion(notifications, language.label)
+        }
+    }
+
+    private fun findLanguage(extension: String, languages: Set<Language>): Language? {
+        return languages.find {
+            it.defaultFileSuffixes.any { suffix ->
+                suffix.equals(extension) || suffix.equals(".$extension")
+            }
         }
     }
 
-    private fun findLanguage(extension: String) = extraEnabledLanguagesInConnectedMode.find {
-        it.defaultFileSuffixes.any { suffix ->
-            suffix.equals(extension) || suffix.equals(".$extension")
+    private fun showAdvancedLanguagePromotion(notifications: SonarLintProjectNotifications, language: String) {
+        val lastModifiedDate: Instant? = getLastModifiedDate()
+
+        if (shouldNotify(lastModifiedDate)) {
+            notifications.notifyAdvancedLanguageSupport(language)
+            Settings.getGlobalSettings().lastPromotionNotificationDate = Instant.now().toEpochMilli()
```

Comment:
I fear that we will end up duplicating the same logic for every type of notification, maybe there is a way to factorize some parts?